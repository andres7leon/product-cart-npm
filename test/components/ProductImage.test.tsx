import React from 'react'
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';


const product1 = {
    id: '1',
    title: 'Coffee Mug - Card',
    img: './coffee-mug.png'
}

describe('Productimage', () => { 
    
    test('debe de mostrar el componente correctaent con la url de la imagen personalizado', () => {
        
        const wrapper = renderer.create(
            <ProductImage img='http://images.jpg' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

    test('debe de mostrar el componente con la imagen por defecto', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product1 } >
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
        
    })
})