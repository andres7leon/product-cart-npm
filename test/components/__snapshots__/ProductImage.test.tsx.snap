// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Productimage debe de mostrar el componente con la imagen por defecto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Productimage debe de mostrar el componente correctaent con la url de la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="http://images.jpg"
/>
`;
